tcpdialer.go

```diff
@@ -422,20 +436,25 @@ func resolveTCPAddrs(addr string, dualStack bool) ([]net.TCPAddr, error) {
 		return nil, err
 	}
 
-	ips, err := net.LookupIP(host)
+	if resolver == nil {
+		resolver = &net.Resolver{}
```

Comment:
```suggestion
		resolver = net.DefaultResolver
```
`net.DefaultResolver` is the default resolver `net.LookupIP` used. This way we don't break anything for people who would overwrite `net.DefaultResolver` instead (which of course is bad practice).